src/deluge/gui/ui/keyboard_screen.cpp

```diff
@@ -430,13 +438,18 @@ void KeyboardScreen::exitAuditionMode() {
 
 void KeyboardScreen::stopAllAuditioning(ModelStack* modelStack, bool switchOffOnThisEndToo) {
 
-	for (int p = 0; p < MAX_NUM_KEYBOARD_PAD_PRESSES; p++) {
-		if (padPresses[p].x != 255) {
-			int noteCode = getNoteCodeFromCoords(padPresses[p].x, padPresses[p].y);
-			((MelodicInstrument*)currentSong->currentClip->output)->endAuditioningForNote(modelStack, noteCode);
-			if (switchOffOnThisEndToo) padPresses[p].x = 255;
+	if (currentSong->currentClip->output->type != INSTRUMENT_TYPE_KIT) {
+		for (int p = 0; p < MAX_NUM_KEYBOARD_PAD_PRESSES; p++) {
+			if (padPresses[p].x != 255) {
+				int noteCode = getNoteCodeFromCoords(padPresses[p].x, padPresses[p].y);
+				((MelodicInstrument*)currentSong->currentClip->output)->endAuditioningForNote(modelStack, noteCode);
+				if (switchOffOnThisEndToo) padPresses[p].x = 255;
+			}
 		}
 	}
+	else {
+		// TODO: Implement for kits
```

Comment:
Maybe flash a `SOON` here?

Reply:
I dont think so, as this function is called mostly from within other routines, so this message would pop up at random-ish times for the user